app/assets/javascripts/discourse/app/static/prosemirror/lib/plugin-utils.js

```diff
@@ -2,6 +2,24 @@ import { InputRule } from "prosemirror-inputrules";
 
 export { getLinkify, isBoundary, isWhiteSpace } from "../lib/markdown-it";
 
+export function buildBBCodeAttrs(attrs, opts = {}) {
```

Comment:
Super nit: as this is not strictly related to ProseMirror specifics/imports, it could also live in a more common place.

The main purpose of these utils in this place is so they can be used in plugins without direct imports when direct imports would mean an eager load of the ProseMirror dependencies.

Reply:
Yep makes sense, will move it